### PR TITLE
📝 Refactor: Readable edition

### PR DESCRIPTION
**Pull request by itsseraphii**
Displays only differences since the folder rename. This PR needs to be approved before 
`refactor` -> `main`

gif

### CODE REVIEW

**Review by DannickB — APPROVED** [review]

File: Core/include/PinExtender.h [hunk 1]
```diff
@@ -6,6 +6,7 @@
 #include <PCA9505_9506.h>
 #include <PCA9685.h>
 
+//Ask Edouard, I don't even know
```
Comment (itsseraphii) [inline Core/include/PinExtender.h:9]: TODO refactor then? J'peux le prendre, c'est wack un peu d'avoir une partie de code que litérallement personne ne comprends lol

File: Core/include/ProxiSensor.h [hunk 2]
```diff
@@ -22,15 +23,46 @@ class ProxiSensor
 public:
     ProxiSensor(Multiplex *muxPtr, int muxAddress);
 
+    /**
+    * @brief Scan each 3 sensors to get the closest distance between the 3 using an average of 3 readings 
+    * @return Closest distance to ground
```
Comment (itsseraphii) [inline Core/include/ProxiSensor.h]: Do we know the unit of measurement? I sure love getting `minDistance = 31`

File: Core/include/ProxiSensor.h [hunk 3]
```diff
@@ -22,15 +23,46 @@ class ProxiSensor
...
+    * @return Trigger distance
+    */
     int GetTriggerDistance();
+    /**
+    * @brief Update brightness and auto calibrate trigger distance with an average of 3 GetMinDistance() readings
+    * @param brightness New brightness lvl
+    */
+    void SetTriggerDistance(bool brightness);
+    /**
+    * @brief Getter for current channel used by proxim's interne MUX
```
Comment (itsseraphii) [inline Core/include/ProxiSensor.h]:
```suggestion
    * @brief Getter for current channel used by proxim's internal MUX 
```

File: Core/include/ProxiSensor.h [hunk 4]
```diff
@@ -22,15 +23,46 @@ class ProxiSensor
...
+    * @return Channel
+    */
     int GetChannel();
+    /**
+    * @brief Setter for channel used by proxim's interne MUX 
+    * @param c Target channel
+    */
     void SetChannel(int c);
+    /**
+    * @brief Print proxim's interne settings, plus it's last min distance read
```
Comment (itsseraphii) [inline Core/include/ProxiSensor.h]:
```suggestion
    * @brief Print proxim's internal settings, plus it's last min distance read
```

File: Core/include/ProxiSensor.h [hunk 5]
```diff
@@ -22,15 +23,46 @@ class ProxiSensor
...
+    */
     void PrintData();
-    void PrintDistance();
+    /**
+    * @brief Set open frame time, lower value increase reading speed but decrease accuracy
+    * @param FrameTiming New frame timing, should be a power of 2
```
Comment (itsseraphii) [inline Core/include/ProxiSensor.h:62]: Should or must? si j'te shoot 13 wha hapen

File: Core/include/bnoHandler.h [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        */
+        bool begin();
+
+        /**
+         * @brief Request data from all BNOs and compute angles between parts
+         */
+        void requestData();
+
+        /**
+         * @brief Print the name and connection status of all BNOs
```
Comment (itsseraphii) [inline Core/include/bnoHandler.h:49]:
```suggestion
         * @brief Print the name and connection status of all BNOs. See EnumBnoPosition for the order.
```

File: Core/include/define.h [hunk 7]
```diff
@@ -107,4 +106,5 @@
 #define SQUAT_MODE 1
 #define FROM_IMU 0
 #define FROM_ENCODER 1
-#define MOTOR_INCREASE 5
\ No newline at end of file
+#define MOTOR_INCREASE 5
+#define PULSES_PER_REVOLUTION 925
```
Comment (itsseraphii) [inline Core/include/define.h:110]: Est-ce qu'on laisse un commentaire juste expliquant d'ou viennent les valeurs? Genre pourquoi 5?

File: Core/include/logic.h [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+public:
+    /**
+     * @brief Compute torque and update required PWM by calling other functions
+     */
+    void Update();
+
+    /**
+     * @brief Cap input value to max/min value
+     * @param val Value tu map (float or int)
```
Comment (itsseraphii) [inline Core/include/logic.h]: Value tu map? I don't understand

File: Core/include/logic.h [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+     * @param max max/min reachable value (float or int)
+     */
+    template <typename T>
+    void LimitMinMax(T &val, T cap);
```
Comment (itsseraphii) [inline Core/include/logic.h:47]: Fancy 👍

File: Core/src/bnoHandler.cpp [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+    angles[static_cast<int> (EnumBnoPosition::KNEE_L)] = abs(tibiaL-thighL);
+
+    angles[static_cast<int> (EnumBnoPosition::HIP_R)] = abs(-thighR-back);
+    angles[static_cast<int> (EnumBnoPosition::KNEE_R)] = abs(tibiaR-thighR);
+
+    angles[static_cast<int> (EnumBnoPosition::EXO_BACK)] = back;
+
+}
+
+// Returns Yaw, same as previous implementation
```
Comment (itsseraphii) [inline Core/src/bnoHandler.cpp]:
```suggestion
// Returns Yaw, same as previous implementation (BNO_055 used on Darianne)
```

File: Core/src/dataCore.cpp [hunk 11]
```diff
@@ -0,0 +1,309 @@
...
+}
+
+void DataCore::setLeftProxi(bool state){
+    leftProxiState = state;
+}
+
+void DataCore::setResetProxim(bool reset){
+    Serial.println("\t proxim reset to high ");
+    needResetProxim = reset;
+}
```
Comment (itsseraphii) [inline Core/src/dataCore.cpp]: Not sure about this comment, it aint spitting facts if I pass it `false`

File: Core/src/logic.cpp [hunk 12]
```diff
@@ -0,0 +1,210 @@
...
+    if (std::is_same<T, int>::val || std::is_same<T, float>::val)
+        val > cap ? val = cap : val < -cap ? val = -cap : 0;
+    else
+        Serial.print("LimitMinMax Error - Invalide data type: ");
+        Serial.println(val);
+    // uncomment if pussy
+    //  if (val > max)
+    //  val = max;
+    // else if (val < -max)
+    // val = -max;
```
Comment (itsseraphii) [inline Core/src/logic.cpp:20]: 😍

**Review by itsseraphii — COMMENTED** [review]
gaming 👍

File: Core/include/PinExtender.h (hunk 1 above)
Comment (DannickB) [reply Core/include/PinExtender.h:9]: yeah make sens, Ed devrait prob s'en charger

File: Core/src/logic.cpp (hunk 12 above)
Comment (DannickB) [reply Core/src/logic.cpp:20]: 😍

File: Core/include/ProxiSensor.h (hunk 5 above)
Comment (DannickB) [reply Core/include/ProxiSensor.h:62]: not much I think, lib docs asked for it tho iirc

File: Core/src/dataCore.cpp (hunk 11 above)
Comment (DannickB) [reply Core/src/dataCore.cpp]: to the shadow realm

**Pull request closed without merging**